### DISCUSSION TITLE
Possible secret key complexity bug

### DISCUSSION THREAD

**Discussion by s0alqasim1d** [bmax121/APatch#317] [Q&A]
I don't know if you consider this a bug or not. I did notice that the key requirements are alphanumeric characters, but I wanted to try a more complex key. So I tried using a `$` at the start of the key and that caused failure stating no key was provided.

I could see the command for a second then it disappears, it seemed indeed that the command had no key provided. Knowing that Unix shell may interpret a string starting with `$` as an environment variable I tried wrapping the key in single quotes which worked. The output then displayed my key enclosed in single quotes when it substituted into the command.

tldr;
If the key is `$secretkey123`, Patching doesn't work. When wrapping in single quotes like: `'$secretkey123'`, patching proceeds successfully.
